### DISCUSSION TITLE
Is there a way to include comments in a config file?

### DISCUSSION THREAD

**Discussion by DiscoRiver** [spf13/viper#1104] [Q&A] [answered]
Hope you're all well!

I'm trying to do some config versioning so I can run config migrations when I update my program. I need to automate things so users don't need to manually add stuff, which is currently the case.

The problem I'm running into is that I need to include comments in the .yaml file, indicating things like permissions and security policy for the file (it's sensitive). I don't see a way to add comments to config items, or a header at the top of the file, and performing `viper.WriteConfig` strips out all comments if I try to include them manually. 

If this isn't possible with Viper, I will create a feature request, but I'm hoping someone can correct me!

**Comment by sagikazarmark** [comment, marked as answer]
Viper does not and cannot support this feature sadly. Since Viper is an abstraction above several configuration sources, supporting comments in a generic way would require abstracting them as well which would be a lot of work with no real benefit.

I'd like to also add that `WriteConfig` is not intended for "updating" config files. That feature was added as a sort of "dump" that can be used for debugging or creating a default config file.

**Reply by DiscoRiver** [reply]
Thanks for the quick reply.

You mentioned that `WriteConfig` isn't intended for updating, so is there an alternative, or a method you'd recommend? Updating a config file seems like it would be within the scope of Viper.

**Reply by sagikazarmark** [reply]
> Updating a config file seems like it would be within the scope of Viper

Well, it isn't. I can see how it would fit into the scope, but Viper is primarily for reading configuration from multiple sources.

If you need to write configuration (especially with preserving comments), then Viper might not be the right tool for you.

**Reply by DiscoRiver** [reply]
Appreciate it, I think I'll just build some custom stuff to update the yaml, and continue to use viper for reading (it's too compatible with Cobra to stop using it). 

All the best.
